sql/planbuilder/parse_test.go

```diff
@@ -45,6 +45,19 @@ func TestPlanBuilder(t *testing.T) {
 	//rewrite = true
 
 	var tests = []planTest{
+		{
```

Comment:
This apparently works in MySQL
```
select t1.x as x, t1.x as x from xy t1, xy t2 order by x;
```


but this fails in MySQL
```
select t1.x as x, t2.x as x from xy t1, xy t2 order by x;
```

I think they both fail in dolt even with your fix.

Also should make sure
```
select x, y as x from xy order by x;
```
still fails

Reply:
good catch, take another peek see if the solution's OK

Reply:
I previously outlined the exact resolution rules for `ORDER BY` in MySQL here: https://github.com/dolthub/dolt/issues/6676